homeassistant/components/raspihats/binary_sensor.py

```diff
@@ -108,6 +109,11 @@ def __init__(self, address, channel, name, invert_logic, device_class):
         self._device_class = device_class
         self._state = self.I2C_HATS_MANAGER.read_di(self._address, self._channel)
 
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        if TYPE_CHECKING:
+            assert self.I2C_HATS_MANAGER
+
```

Comment:
We need to register the callbacks in the executor as the registration waits for a threading lock.